### PR TITLE
HHH-12597 - Support Hibernate dialect for InterSystems IRIS Data Platform

### PR DESCRIPTION
**Pull request by InterSystemsCorporation**
https://hibernate.atlassian.net/browse/HHH-12597

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+
+	/**
+	 * Creates new <code>InterSystemsIRISDialect</code> instance. Sets up the JDBC /
+	 * Cach&eacute; type mappings.
+	 */
+	public InterSystemsIRISDialect() {
+		super();
+		commonRegistration();
+		registerFunctions();
+		this.limitHandler = new TopLimitHandler(true, true);
```
Comment (Naros) [inline hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java]: Please make sure you're placing spaces after `(` and before `)`.

File: hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java [hunk 2]
```diff
@@ -0,0 +1,603 @@
...
+		registerColumnType( Types.REAL, "real" );
+		registerColumnType( Types.SMALLINT, "smallint" );
+		registerColumnType( Types.TIMESTAMP, "timestamp" );
+		registerColumnType( Types.TIME, "time" );
+		registerColumnType( Types.TINYINT, "tinyint" );
+		registerColumnType( Types.VARBINARY, "longvarbinary" );
+		registerColumnType( Types.VARCHAR, "varchar($l)" );
+		registerColumnType( Types.BLOB, "longvarbinary" );
+		registerColumnType( Types.CLOB, "longvarchar" );
+		registerColumnType(Types.BOOLEAN, "integer");
```
Comment (Naros) [inline hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java]: Another place here spaces after `(` and before `)` are needed.

File: hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java [hunk 3]
```diff
@@ -0,0 +1,603 @@
...
+
+	}
+
+	protected final void registerFunctions() {
+		this.registerFunction( "str", new VarArgsSQLFunction( StandardBasicTypes.STRING, "str(", ",", ")" ) );
+		//new overwrites
+		registerFunction( "year", new StandardSQLFunction( "year", StandardBasicTypes.INTEGER ) );
+		registerFunction( "sqrt", new StandardSQLFunction( "sqrt", StandardBasicTypes.DOUBLE ) );
+		registerFunction( "log10", new StandardJDBCEscapeFunction( "log10", StandardBasicTypes.DOUBLE ) );
+		registerFunction( "current_timestamp", new NoArgSQLFunction( "current_timestamp", StandardBasicTypes.TIMESTAMP, false ));
```
Comment (Naros) [inline hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java]: Missing extra space before last `)`.

File: hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java [hunk 4]
```diff
@@ -0,0 +1,603 @@
...
+
+	// lock acquisition support ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+	@Override
+	public boolean supportsOuterJoinForUpdate() {
+		return false;
+	}
+
+	@Override
+	public LockingStrategy getLockingStrategy(Lockable lockable, LockMode lockMode) {
```
Comment (Naros) [inline hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java]: This method could use some formatting fixups.

File: hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java [hunk 5]
```diff
@@ -0,0 +1,603 @@
...
+	public boolean supportsResultSetPositionQueryMethodsOnForwardOnlyCursor() {
+		return false;
+	}
+
+	/**
+	 * ddl like ""value" integer null check ("value">=2 AND "value"<=10)" isn't supported
+	 */
+	@Override
+	public boolean supportsColumnCheck()
+	{
```
Comment (Naros) [inline hibernate-core/src/main/java/org/hibernate/dialect/InterSystemsIRISDialect.java]: Begin braces, `{` should begin on the line where the method is defined, not on a new-line.
There are several occurrences of this in this file.

**Comment by Naros** [comment]
I also notice you want to merge this into branch 5.2.  
Please rebase your PR and submit it for inclusion against master and not 5.2.

**Comment by vladmihalcea** [comment]
@InterSystemsCorporation Traditionally, only if the community requests a certain Dialect, it will be included in the `hibernate-core`.

A Dialect works even if it is included via an external dependency.

**Comment by InterSystemsCorporation** [comment]
We are rebasing our changes against **master (5.3)**.

When complete, we will submit a new pull request.

**Pull request closed without merging**